src/overlays/effects/ovl_Effect_Ss_Kakera/z_eff_ss_kakera.c

```diff
@@ -8,50 +8,355 @@
+        } else {
+            gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[this->regs[11]].segment);
+        }
+    }
+
+    Matrix_InsertTranslation(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+    Matrix_RotateY(this->regs[3], MTXMODE_APPLY);
+    Matrix_InsertXRotation_s(this->regs[2], MTXMODE_APPLY);
+    Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+
+    if ((((this->regs[4] >> 7) & 1) << 7) == 0x80) {
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        if (colorIndex >= 0) {
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, D_8097EAD8[colorIndex].lod, D_8097EAD8[colorIndex].color.r,
+                            D_8097EAD8[colorIndex].color.g, D_8097EAD8[colorIndex].color.b, 255);
+        }
+        gSPDisplayList(POLY_XLU_DISP++, this->gfx);
+    } else {
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        func_8012C28C(globalCtx->state.gfxCtx);
+        if (colorIndex >= 0) {
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, D_8097EAD8[colorIndex].lod, D_8097EAD8[colorIndex].color.r,
+                            D_8097EAD8[colorIndex].color.g, D_8097EAD8[colorIndex].color.b, 255);
+        }
+        gSPDisplayList(POLY_OPA_DISP++, this->gfx);
+    }
+    CLOSE_DISPS(gfxCtx);
+}
+
+void func_8097E130(EffectSs* this, GlobalContext* globalCtx) {
+    this->regs[11] = Object_GetIndex(&globalCtx->objectCtx, this->regs[10]);
+    if ((this->regs[11] < 0) || (!Object_IsLoaded(&globalCtx->objectCtx, this->regs[11]))) {
+        this->life = 0;
+        this->draw = NULL;
+    }
+}
+
+void func_8097E19C(EffectSs* this) {
+    f32 range;
+    f32 temp_f12;
+    f32 temp_f16;
+    f32 temp_f2;
+    f32 temp_f18;
+    f32 temp_f0;
+    f32 temp_f20;
+    f32 temp_f2_2;
+
+    temp_f18 = this->regs[5] / 1024.0f;
+    temp_f20 = this->regs[6] / 1024.0f;
+    range = (this->regs[9] / 1024.0f) * 4.0f;
+    temp_f2 = this->velocity.x - func_8097DE30(0.0f, range);
+    temp_f16 = this->velocity.y - func_8097DE30(0.0f, range);
+    temp_f12 = this->velocity.z - func_8097DE30(0.0f, range);
+    temp_f0 = temp_f2 * temp_f18;
+    temp_f2_2 = SQ(temp_f2) * temp_f20;
+    if (temp_f2 > 0.0f) {
+        this->velocity.x -= (temp_f0 + temp_f2_2);
+    } else {
+        this->velocity.x -= (temp_f0 - temp_f2_2);
+    }
+    temp_f0 = temp_f16 * temp_f18;
+    temp_f2_2 = SQ(temp_f16) * temp_f20;
+    if (temp_f16 > 0.0f) {
+        this->velocity.y -= (temp_f0 + temp_f2_2);
+    } else {
+        this->velocity.y -= (temp_f0 - temp_f2_2);
+    }
+    if (temp_f12 > 0.0f) {
+        this->velocity.z -= (temp_f0 + temp_f2_2);
+    } else {
+        this->velocity.z -= (temp_f0 - temp_f2_2);
+    }
+}
+
+void func_8097E34C(EffectSs* this) {
+    this->accel.z = 0.0f;
+    this->accel.y = 0.0f;
+    this->accel.x = 0.0f;
```

Comment:
```suggestion
    this->accel.x = this->accel.y = this->accel.z = 0.0f;
```
would this work? (Not that it makes a huge difference)

Reply:
yep, works there and a few other places

Reply:
I don't have any real preference. You could do a one-liner, or separated, I was just curious lol

Reply:
We know that they do use chained assigns, so it's probably better to do it.